src/client/entity-editor/common/entity-search-field-option.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2015-2017  Ben Ockmore
+ *               2015-2016  Sean Burke
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import CustomInput from '../../input';
+import Entity from './entity';
+import LinkedEntity from './linked-entity';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Async as SelectAsync} from 'react-select';
+import ValidationLabel from '../common/validation-label';
+import _ from 'lodash';
+import makeImmutable from './make-immutable';
+import request from 'superagent-bluebird-promise';
+
+
+const ImmutableAsyncSelect = makeImmutable(SelectAsync);
+
+class EntitySearchFieldOption extends React.Component {
+	constructor(props) {
+		super(props);
+
+		// React does not autobind non-React class methods
+		this.fetchOptions = this.fetchOptions.bind(this);
+		this.isArea = this.isArea.bind(this);
+		this.entityToOption = this.entityToOption.bind(this);
+	}
+
+	/**
+	 * Determines whether an entity provided to the EntitySearch component is an
+	 * Area, using the present attributes.
+	 *
+	 * @param {Object} entity the entity to test
+	 * @returns {boolean} true if the entity looks like an Area
+	 */
+	isArea(entity) {
+		if (entity.type === 'Area') {
+			return true;
+		}
+
+		if (entity.gid) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Takes an entity and converts it to a format acceptable to react-select.
+	 *
+	 * @param {Object} entity the entity to convert
+	 * @returns {Object} the formatted data
+	 * @param {Array<LanguageOption>} languageOptions - The list of possible languages for an
+	 * entity.
+	 */
+	entityToOption(entity) {
+		const id = this.isArea(entity) ? entity.id : entity.bbid;
+		const languageId = _.get(entity, ['defaultAlias', 'languageId']);
+		const language = this.props.languageOptions.find(index => index.value === languageId).label;
```

Comment:
This will throw an error if `.find` does not return anything.
It's unlikely, but it's worth making sure it doesn't happen. For example, below in the return value you can return `language: language && language.label`.

Reply:
okay.